conans/paths/package_layouts/package_cache_layout.py

```diff
@@ -110,6 +110,43 @@ def package(self, pref):
         assert pref.ref == self._ref, "{!r} != {!r}".format(pref.ref, self._ref)
         return os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
 
+    @contextmanager
+    def set_dirty_context_manager(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        set_dirty(pkg_folder)
+        yield
+        clean_dirty(pkg_folder)
+
+    def package_is_dirty(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        return is_dirty(pkg_folder)
+
+    def package_set_dirty(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        return set_dirty(pkg_folder)
+
+    def package_clean_dirty(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        return clean_dirty(pkg_folder)
+
+    def package_id_exists(self, package_id):
```

Comment:
There is a similar ``package_exists()`` method, but I didn't want to change it. That one is doing checks of the revisions, for the ``conan remove`` functionality, better not to mess with it.

Reply:
Nevertheless, what happens if I remove manually the short_paths folder? In `develop` this function would return False (it was checking the short-path), now it will return True (checking the one in the cache)